### PR TITLE
870 build uki

### PR DESCRIPTION
**Pull request by jimmykarily**
Use the following command to easily test:

```
CGO_ENABLED=false go build -o build/enki && docker run --rm --entrypoint /bin/enki -v $PWD/build/enki:/bin/enki  -v $PWD/build:/result -v $PWD/e2e/assets/keys/:/keys enki-image build-uki quay.io/kairos/fedora:38-standard-amd64-generic-v2.5.0-k3sv1.28.5-k3s1  /result/fedora.uki.iso /keys
```

It assumes an enki-image has been built using the `e2e/assets/enki-Dockerfile` dockerfile.

Part of: https://github.com/kairos-io/kairos/issues/870

### CODE REVIEW

File: pkg/action/build-uki.go [hunk 1]
```diff
@@ -0,0 +1,538 @@
...
+			return fmt.Errorf("writing record %q failed: %w", filePath, err)
+		}
+
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("error walking the source dir: %w", err)
+	}
+
+	if err := cpio.WriteTrailer(rw); err != nil {
```
Comment (jimmykarily) [inline pkg/action/build-uki.go:217]: Commands in this method were mostly copied from here: https://github.com/u-root/u-root/blob/25faeb4566c826ea11bc725f291af8f46d893cc5/cmds/core/cpio/cpio.go#L133-L145

File: Earthfile [hunk 2]
```diff
@@ -13,7 +18,7 @@ test:
     ARG LABEL_FILTER=
     ENV CGO_ENABLED=1
     # Some test require the docker sock exposed
-    WITH DOCKER
+    WITH DOCKER --load enki-image=(+enki-image)
```
Comment (jimmykarily) [inline Earthfile:21]: NOTE: We can do this trick on `kairos-io/kairos` too in order to load the `+base-image` into the target in which we are planning to run enki on.

**Comment by Itxaka** [comment]
>  check with Itxaka what was the intermediate uki file needed for, and if we still need it we need to refactor

There is several targets in earthfile:

 - uki-base: Just generates the artifacts needed to build uki (kernel,initramfs,cmdline,etc...)
 - uki-build: This gets all the artifacts and generates the final efi files with signatures and everything
 - uki-image-artifacts: This just creates the expected paths and copies the files in the proper places
 - uki-image: This generates the final container image with the artifact from uki-image-artifacts. Needs two steps because this image has a base image of SCRATCH thus it cannot mkdir anything. So this and uki-image-artifacts could be mertged into one target, but the base image has to be SCRATCH to not ship anything else in the final image.
 - uki-iso: Generates the iso from the uki-build artifacts


In our case, as we have more control over this and we have 2 clear outputs (iso and container image) it should not be needed as we dont work with the earthly constraints.

so basically build artifacts -> output container image or ISO

File: cmd/build-uki.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+			"ResultFile - the path to the resulting iso file\n" +
+			"KeysDirectory - the path to the directory with the signing keys.\n" +
+			"    The following files are expected in this directory:\n" +
+			"    - DB.crt\n" +
+			"    - DB.der\n" +
+			"    - DB.key\n" +
+			"    - KEK.der\n" +
+			"    - PK.der\n" +
+			"    - tpm2-pcr-private.pem\n",
+		Args: cobra.ExactArgs(3),
```
Comment (Itxaka) [inline cmd/build-uki.go:33]: args should be options flags in some cases IMO.

output -> if not marked just use a generic output.iso
keys -> if not set, just output a uki with no signatures? Should we allow doing that for testing purpouses so people can test before inserting their real keys?

File: pkg/action/build-uki.go [hunk 4]
```diff
@@ -0,0 +1,538 @@
...
+
+	"github.com/u-root/u-root/pkg/cpio"
+	"golang.org/x/exp/maps"
+
+	"github.com/kairos-io/enki/pkg/types"
+	"github.com/kairos-io/kairos-agent/v2/pkg/elemental"
+	v1 "github.com/kairos-io/kairos-agent/v2/pkg/types/v1"
+)
+
+const Cmdline = "console=ttyS0 console=tty1 net.ifnames=1 rd.immucore.oemlabel=COS_OEM rd.immucore.debug rd.immucore.oemtimeout=2 rd.immucore.uki selinux=0"
```
Comment (Itxaka) [inline pkg/action/build-uki.go:23]: this should be user overridable IMHO. Nice to have and we ALWAYS would need to set things like `rd.immucore.oemlabel=COS_OEM rd.immucore.oemtimeout=2 rd.immucore.uki` but the rest can be user set I think

File: pkg/action/build-uki.go [hunk 5]
```diff
@@ -0,0 +1,538 @@
...
+		"sbsign",
+		"dd",
+		"mkfs.msdos",
+		"mmd",
+		"mcopy",
+		"xorriso",
+	}
+
+	neededFiles := []string{
+		"/usr/lib/systemd/boot/efi/linuxx64.efi.stub",
```
Comment (Itxaka) [inline pkg/action/build-uki.go:114]: this also needs change depending on the arch we are building for, this is for x86, arm64 has different stub+boot :(

**Review by Itxaka — APPROVED** [review]

**Comment by Itxaka** [comment]
good enough for alpha IMHO, lets merge and improve
